pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -239,45 +239,47 @@ t_slice new_slice(int32_t start, int32_t end, int32_t step)
     return (slice);
 }
 
-t_ndarray array_slicing(t_ndarray arr, int n, ...)
+t_ndarray array_slicing(t_ndarray arr, int n, int n_slices, ...)
 {
     t_ndarray view;
     va_list  va;
     t_slice slice;
     int32_t start = 0;
+    int32_t offset = 0;
 
     view.nd = n;
     view.type = arr.type;
     view.type_size = arr.type_size;
-    view.shape = malloc(sizeof(int64_t) * arr.nd);
-    view.strides = malloc(sizeof(int64_t) * arr.nd);
-    memcpy(view.strides, arr.strides, sizeof(int64_t) * arr.nd);
+    view.shape = malloc(sizeof(int64_t) * view.nd);
+    view.strides = malloc(sizeof(int64_t) * view.nd);
     view.is_view = true;
-    va_start(va, n);
-    for (int32_t i = 0; i < arr.nd ; i++)
+    
+    va_start(va, n_slices);
+    for (int32_t i = 0; i < n_slices; i++)
     {
         slice = va_arg(va, t_slice);
-        view.shape[i] = (slice.end - slice.start + (slice.step - 1)) / slice.step; // we need to round up the shape
         start += slice.start * arr.strides[i];
-        view.strides[i] *= slice.step;
     }
     va_end(va);
-    int32_t j = arr.nd - view.nd;
-    if (j)
+
+    va_start(va, n_slices);
+    for (int32_t i = 0; i < n_slices; i++)
     {
-        int64_t *tmp_strides = malloc(sizeof(int32_t) * view.nd);
-        int64_t *tmp_shape = malloc(sizeof(int32_t) * view.nd);
-        for (int32_t i = 0; i < view.nd; i++)
+        slice = va_arg(va, t_slice);
+        if (slice.end - slice.start > 1 && !offset)
         {
-            tmp_strides[i] = view.strides[j];
-            tmp_shape[i] = view.shape[j];
-            j++;
+            offset = i;
+            break ;
         }
-        free(view.shape);
-        free(view.strides);
-        view.strides = tmp_strides;
-        view.shape = tmp_shape;
     }
+    for (int32_t i = 0; i < n; i++)
+    {
+        view.shape[i] = (slice.end - slice.start + (slice.step - 1)) / slice.step;
+        view.strides[i] = arr.strides[offset + i] * slice.step;
+        slice = va_arg(va, t_slice);
+    }
```

Comment:
I get the impression that this only handles consecutive removed dimensions. I suspect something like `[:,0,:,2,:]` would not work

Reply:
```python3
if __name__ == "__main__":
     x1 = np.ones((2, 3, 4, 5, 6), dtype=int)
     a = x1[:,0,:,2,:]
     print(1)
```
I tried this test and it worked fine. It gave the same result as python.